CI: lychee link checker fails on localhost:8000 in CONTRIBUTING.md

## Problem

The template-generated `CONTRIBUTING.md` contains a bare URL `<http://localhost:8000>` (line referencing local docs preview). The lychee link checker CI job tries to resolve this and fails with:

```
[ERROR] http://localhost:8000/ | Network error: Connection refused
```

## Root Cause

`.lychee.toml` has `exclude_all_private = true` which covers private IP ranges (10.x, 172.16.x, 192.168.x) but does **not** exclude `localhost` by hostname.

## Fix Options

1. Add `'^http://localhost'` to the `exclude` list in `.lychee.toml`
2. Or rewrite the CONTRIBUTING.md line to not use a bare URL (e.g. `localhost:8000` without angle brackets)

Option 1 is preferred since other localhost references may appear in docs.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -26,6 +26,8 @@ jobs:
 
   changes:
     runs-on: blacksmith-4vcpu-ubuntu-2404
+    permissions:
+      pull-requests: read
     outputs:
       src: ${{ steps.filter.outputs.src }}
     steps:

--- a/.lychee.toml
+++ b/.lychee.toml
@@ -19,6 +19,8 @@ exclude = [
     'discussions/new\?category=',
     # PyPI package URL with Jinja placeholder renders as bare /p/
     '^https://pypi\.org/p/$',
+    # Localhost URLs (e.g. docs preview in CONTRIBUTING.md)
+    '^http://localhost',
 ]
 
 # Exclude local/private addresses

--- a/project/.github/workflows/ci.yml.jinja
+++ b/project/.github/workflows/ci.yml.jinja
@@ -32,6 +32,8 @@ jobs:
 {%- else %}
     runs-on: ubuntu-latest
 {%- endif %}
+    permissions:
+      pull-requests: read
     outputs:
       src: {% raw %}${{ steps.filter.outputs.src }}{% endraw %}
     steps:

--- a/project/.lychee.toml.jinja
+++ b/project/.lychee.toml.jinja
@@ -11,6 +11,8 @@ exclude = [
     '^https://pypi\.org/project/my-',
     '^https://codecov\.io/gh/your-',
     '^mailto:',
+    # Localhost URLs (e.g. docs preview in CONTRIBUTING.md)
+    '^http://localhost',
 ]
 
 # Exclude local/private addresses

--- a/project/pyproject.toml.jinja
+++ b/project/pyproject.toml.jinja
@@ -233,6 +233,7 @@ tags = "sh -c 'git fetch --tags && git tag --sort=-version:refname'"
 # GitHub utilities
 releases = "gh release list --limit 10"
 runs = "gh run list --limit 10"
+checks = "gh pr checks --watch"
 watch = "gh run watch"
 {%- endif %}
 

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -157,6 +157,7 @@ prek = "prek run --all-files"
 releases = "gh release list --limit 10"
 tags = "sh -c 'git fetch --tags && git tag --sort=-version:refname'"
 runs = "gh run list --limit 10"
+checks = "gh pr checks --watch"
 watch = "gh run watch"
 
 [tool.coverage.run]

